src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -48,12 +48,15 @@ public static void generateReposReport(List<RepoConfiguration> configs, String o
         FileUtil.copyTemplate(is, outputPath);
 
         for (RepoConfiguration config : configs) {
-            Path repoReportDirectory = Paths.get(outputPath, config.getDisplayName());
+            Path repoReportDirectory;
             try {
-                FileUtil.createDirectory(repoReportDirectory);
                 GitDownloader.downloadRepo(config);
+                repoReportDirectory = Paths.get(outputPath, config.getDisplayName());
+                FileUtil.createDirectory(repoReportDirectory);
             } catch (GitDownloaderException gde) {
                 logger.log(Level.WARNING, "Exception met while trying to clone the repo, will skip this one", gde);
+                repoReportDirectory = Paths.get(outputPath, config.getDisplayName());
+                FileUtil.createDirectory(repoReportDirectory);
```

Comment:
Report directory can only be created after the display name has been updated using the current working branch i.e. default branch HEAD is used when cloning the repository after which the local current working branch when the analysis is performed against a local repository is retrieved and display name is updated accordingly with the current branch name i.e. from testAlpha_HEAD to testAlpha_test . I've discussed this change with the previous reviewer in detail so you can refer to that if you need any additional information about this change.

Reply:
We could do the update and then create the directory before the `try` and `catch` clause right, rather than having multiple lines of this same code for each exception caught.

Reply:
what do u mean? the reporeportdirectory path and directory creation can only be done after the repo has been downloaded i.e. after line ```GitDownloader.downloadRepo(config);``` which is inside the try-catch therefore it cannot be moved to before the try-catch clause

Reply:
>the repoReportDirectory path and directory creation can only be done after the repo has been downloaded i.e. after line GitDownloader.downloadRepo(config)

Sorry, forgot that the branch is only updated after downloading the repo =x